ibc_public/utils_connectivity.py

```diff
@@ -75,82 +213,546 @@ def get_ses_modality(task):
+    # make predictions for the left-out test subjects
+    predictions = classifier.predict(connectomes[test])
+    accuracy = accuracy_score(classes[test], predictions)
+    if classify in ["Runs", "Subjects"]:
+        auc = 0
+    else:
+        auc = roc_auc_score(
+            classes[test], classifier.decision_function(connectomes[test])
+        )
+    # store the scores for this cross-validation fold
+    results = _update_results(
+        results,
+        accuracy,
+        auc,
+        predictions,
+        classes[test],
+        train,
+        test,
+        task_label,
+        connectivity_measure,
+        classify,
+    )
 
-    if task_modality == "functional":
-        ses_mod_dir = "func"
-    elif task_modality == "structural":
-        ses_mod_dir = "dwi"
-
-    all_subject_connectivity = []
-
-    for sub, sess in sub_ses.items():
-        # if it is pearsons correlation, we calculated it for each run
-        if conn_measure.split("_")[0] == "Pearsons":
-            sess = sorted(sess)
-            subject_connectivity = []
-            for ses in sess:
-                file_loc = os.path.join(data_root, sub, ses, ses_mod_dir)
-                runs = glob(os.path.join(file_loc, f"*{conn_measure}*"))
-                assert len(runs) > 0
-                for run in runs:
-                    pearson_mat = pd.read_csv(
-                        run, index_col=False, header=None
-                    )
-                    pearson_mat_flat = connectome.sym_matrix_to_vec(
-                        pearson_mat.to_numpy(), discard_diagonal=True
-                    )
-                    subject_connectivity.append(pearson_mat_flat)
-            if average_runs:
-                subject_connectivity = np.mean(subject_connectivity, axis=0)
-        # if it is any other connectivity measure, we have only one estimate
-        # over all runs
-        else:
-            if task_modality == "functional":
-                sess = sorted(sess)
-                ses = sess[-1]
-            elif task_modality == "structural":
-                ses = sess
-            file_loc = os.path.join(data_root, sub, ses, ses_mod_dir)
-            subject_connectivity = glob(
-                os.path.join(file_loc, f"*{conn_measure}*")
-            )
-            assert len(subject_connectivity) == 1
-            subject_connectivity = pd.read_csv(
-                subject_connectivity[0], index_col=False, header=None
+    return results
+
+
+def cross_validate(
+    connectomes,
+    classes,
+    splits,
+    task_label,
+    connectivity_measure,
+    results,
+    classify,
+):
+    """Cross validate the given connectomes using the given classes.
+
+    Parameters
+    ----------
+    connectomes : numpy array
+        Array containing the connectomes to classify. Each row is a vectorised connectome.
+    classes : numpy array
+        Array containing the class for each connectome.
+    splits : numpy array
```

Comment:
or list of arrays ? Can you maybe clarif the expected structure, array dimension, etc.